fvm/environment/env.go

```diff
@@ -102,12 +102,11 @@ type EnvironmentParams struct {
 }
 
 func DefaultEnvironmentParams() EnvironmentParams {
-	const chainID = flow.Mainnet
 	return EnvironmentParams{
-		Chain:                 chainID.Chain(),
+		Chain:                 flow.Mainnet.Chain(),
```

Comment:
Not related to this PR, but something I noticed while adding the fix: Since environment params default the chain to `mainnet`, when we set the chainID later to the context with `fvm.WithChain` option, should we also update the `EnvironmentParams.Params` to the given value as well?
i.e: in the following place https://github.com/onflow/flow-go/blob/e7dd12ef3dbb098503400d92db1011f83a47899a/fvm/context.go#L77-L82

Should we also add:
```go
ctx.EnvironmentParams.Chain = chain
```
Or is the `DefaultEnvironmentParams` only used in tests?

Reply:
`ctx.Chain` is equivalent to `ctx.EnvironmentParams.Chain`. `environment.EnvironmentParams` is an embedded field of `Context`. So it should be good now.

The problem was that previously the runtime params also depended on `ChainID` in which case if you changed the chain you also should have changed the runtime params.

Reply:
ah right, it's an embedded field 👍